clients/src/main/java/org/apache/kafka/clients/admin/DeleteRecordsResult.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Map;
+
+@InterfaceStability.Evolving
+public class DeleteRecordsResult {
+
+    private final Map<TopicPartition, KafkaFuture<Long>> futures;
+
+    DeleteRecordsResult(Map<TopicPartition, KafkaFuture<Long>> futures) {
+        this.futures = futures;
+    }
+
+    /**
+     * Return a map from topic partition to futures which can be used to check the status of
+     * individual deletions.
+     */
+    public Map<TopicPartition, KafkaFuture<Long>> values() {
```

Comment:
`values` is a very generic name.  How about calling this `lowWatermarks`?

Reply:
@cmccabe I agree with you but I'm following what seems to be the common pattern even for other results in the Admin Client. For example, `DeleteAclsResult`, `DescribeConfigsResult`, `DescribeReplicaLogDirsResult`, `DescribeTopicsResult`. It seems that only `ListTopicsResult` and `DescribeClusterResult` are using meaningful names. I'm happy to change here from `values` to `lowWatermarks` but what do you think if I create a JIRA and a corresponding PR for giving meaningful names to the corresponding methods for above classes as well ?